astropy/utils/tests/test_decorators.py

```diff
@@ -406,22 +383,22 @@ def test_deprecated_argument_multi_deprecation():
     def test(a, b, c):
         return a, b, c
 
-    with catch_warnings(AstropyDeprecationWarning) as w:
+    with pytest.warns(AstropyDeprecationWarning) as w:
         assert test(x=1, y=2, z=3) == (1, 2, 3)
-        assert len(w) == 3
+    assert len(w) == 3
 
     # Make sure relax is valid for all arguments
-    with catch_warnings(AstropyUserWarning) as w:
+    with pytest.warns(AstropyUserWarning) as w:
         assert test(x=1, y=2, z=3, b=3) == (1, 3, 3)
-        assert len(w) == 1
+    assert len(w) == 4
```

Comment:
Not sure what `catch_warnings` was doing but now the test emits more warnings, which makes more sense because L388 above got 3 for 3, so here is should have been 4 for 4.